Task3/scripts/models/minesweeper.js

```diff
@@ -0,0 +1,129 @@
+class Minesweeper {
+    constructor() {
+        this.height = 0;
+        this.width = 0;
+        this.bombProbability = 0;
+
+        this.mineField = [];
+        this.numberOfBombs = 0;
+    }
+
+    startNewGame(options) {
+        if (options != null) {
+
+            if (options.height != null && options.width != null) {
+                this.height = options.height;
+                this.width = options.width;
+            }
+            
+            if(options.bombProbability != null){
+                this.bombProbability = options.bombProbability;
+            }
+        }
+
+        this.mineField = [];
+        this.numberOfBombs = 0;
+
+        this._checkModelState();
+        this._generateMineField();
+        this._calculateBombs();
+    }
+
+    getNeighborCellsCoords(xPos, yPos) {
+        var neighbors = [];
+        const CELLS_COUNT = 9;
+
+        for (let i = 0; i < CELLS_COUNT; i++) {
+            if (i === 4) {
+                continue; // dont't return target cell's coords.
+            }
+
+            let x = xPos + i % 3 - 1;
+            let y = yPos + Math.floor(i / 3) - 1;
+
+
+            if (this.getCellXY(x, y) == null) {
+                continue;
+            }
+
+            neighbors.push({
+                xPos: x,
+                yPos: y
+            });
+        }
+
+        return neighbors;
+    }
+
+    getCellXY(xPos, yPos) {
+        if (this.mineField[yPos] == null || this.mineField[yPos][xPos] == null) {
+            return null;
+        }
+        return this.mineField[yPos][xPos];
+    }
+
+    _checkModelState() {
+        if (isNaN(this.height) || this.height <= 0) {
```

Comment:
I competed my tests with infinity values tests and got out of memory exception. It's not what I expected ) I gonna fix it, thaks a lot.